src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.cs

```diff
@@ -1029,6 +1034,7 @@ private void VisitStatement(IOperation operation)
             IOperation saveCurrentStatement = _currentStatement;
             _currentStatement = operation;
             Debug.Assert(_evalStack.Count == 0);
+            Debug.Assert(_currentInitializedInstance == null);
```

Comment:
> Debug.Assert(_currentInitializedInstance == null); [](start = 11, length = 51)

What happens if the object initializer is a lambda expression? I assume that will cause us to come back to VisitStatement.

Reply:
No. While support isn't yet implemented, our decision for lambdas and local functions is that they will be separate graphs, and not put into the main graph. Once you're in an expression, nothing should cause VisitStatement to be called again until you exit that expression.

---
In reply to: [185932789](https://github.com/dotnet/roslyn/pull/26389#discussion_r185932789) [](ancestors = 185932789)